tdp/conftest.py

```diff
@@ -16,31 +18,79 @@
     PLAYBOOKS_DIRECTORY_NAME,
     YML_EXTENSION,
 )
-from tdp.core.models import BaseModel
-
-DATABASE_URL = "sqlite:///:memory:"
+from tdp.core.models import BaseModel, init_database
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    parser.addoption(
+        "--database-dsn",
+        dest="database_dsn",
+        action="append",
+        default=["sqlite"],
+        help="Add database DSN.",
+    )
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
+    """Pytest hook to generate tests based on the database dsn option."""
+    if "db_dsn" in metafunc.fixturenames:
+        metafunc.parametrize(
+            "db_dsn", metafunc.config.getoption("database_dsn"), indirect=True  # type: ignore
+        )
+
+
+@pytest.fixture
+def db_dsn(
+    request: pytest.FixtureRequest, tmp_path: Path
+) -> Generator[str, None, None]:
+    """Return a database dsn.
+
+    Ensure that the database is cleaned up after each test is done.
+
+    We create a temp path instead of the default in-memory sqlite database as some test
+    need to generate several engine instances (which will loose the data between them).
+    Concerned tests are CLI tests that need to perform a `tdp init` at the beginning of
+    the test.
+    """
+    database_dsn = cast(str, request.param)
+    # Assign a temporary database for sqlite
+    if database_dsn == "sqlite":
+        database_dsn = f"sqlite:///{tmp_path / 'test.db'}"
+    yield database_dsn
+    # Drop the database content
+    engine = create_engine(database_dsn)
+    BaseModel.metadata.drop_all(engine)
+    engine.dispose()
 
 
-# TODO: This fixture should return a database dsn
 @pytest.fixture()
-def db_session() -> Generator[Session, None, None]:
-    # Connect to the database
-    engine = create_engine(DATABASE_URL)
-
-    # Create tables
-    BaseModel.metadata.create_all(engine)
+def db_engine_uninitialized(db_dsn: str) -> Generator[Engine, None, None]:
+    """Create a database engine."""
+    engine = create_engine(db_dsn)
+    yield engine
+    engine.dispose()
 
-    # Create a session
-    session_maker = sessionmaker(bind=engine)
-    session = session_maker()
-    yield session
 
-    # Close and rollback for isolation
-    session.close()
-    BaseModel.metadata.drop_all(engine)
+@pytest.fixture()
+def db_engine_initialized(db_dsn: str) -> Generator[Engine, None, None]:
```

Comment:
Why not use the fixture `db_engine_uninitialized` as parameter ?

Reply:
You're right, didn't thought about it.

Reply:
Done